### PR TITLE
:sparkles: UPDATE Action

### PR DESCRIPTION
**Pull request by guillaumecrespel**

### CODE REVIEW

File: src/middlewares/core.js [hunk 1]
```diff
@@ -31,6 +31,15 @@ const coreReducer = key => prefix =>
           initialized: true,
         }
       }
+      case UPDATE(prefix): {
+        const sameKey = state.array.find(o => o[key] === payload[key])
```
Comment (fabienjuif) [inline src/middlewares/core.js]: `const instance = ...`

Comment (fabienjuif) [inline src/middlewares/core.js]: `const instanceKey = payload[key]`

File: src/middlewares/core.js [hunk 2]
```diff
@@ -31,6 +31,15 @@ const coreReducer = key => prefix =>
...
+        if (!sameKey) return state
+        return {
+          ...state,
+          data: { ...state.data, [payload[key]]: { ...state.data[payload[key]], ...payload } },
```
Comment (fabienjuif) [inline src/middlewares/core.js]: `data: { ...state.data, [instanceKey]: { ...state.data[instanceKey], ...payload } }`

Comment (guillaumecrespel) [inline src/middlewares/core.js]: :ok_hand: @fabienjuif

File: src/middlewares/core.js (hunk 1 above)
Comment (guillaumecrespel) [inline src/middlewares/core.js]: :ok_hand: @fabienjuif

File: src/middlewares/core.spec.js [hunk 3]
```diff
@@ -64,4 +64,18 @@ describe('middlewares/core', () => {
...
   ).toMatchSnapshot())
+
+  it('should update an element [elm3]', () => expect(
+    testPrefix({
+      state,
+      action: update(prefix)({ ...Element('elm3'), some: 'other UPDATE' }),
+    }),
+  ).toMatchSnapshot())
+
+  it('should update an element not present [elm12]', () => expect(
```
Comment (guillaumecrespel) [inline src/middlewares/core.spec.js]: :ok_hand: @fabienjuif

File: src/middlewares/core.js [hunk 4]
```diff
@@ -32,12 +32,12 @@ const coreReducer = key => prefix =>
         }
       }
       case UPDATE(prefix): {
-        const sameKey = state.array.find(o => o[key] === payload[key])
-        if (!sameKey) return state
+        const instanceKey = payload[key]
+        if (!instanceKey) return state
```
Comment (fabienjuif) [inline src/middlewares/core.js]: Your diff show that you are not doing same control than before

**Pull request merged**